### PR TITLE
Rianhughes/07 update traces

### PR DESCRIPTION
**Pull request by rianhughes**

### CODE REVIEW

**Comment by rianhughes** [comment]
Implement EXECUTION_RESOURCES and COMPUTATION_RESOURCES
Update TRANSACTION_TRACE, DECLARE_TXN_TRACE, DEPLOY_ACCOUNT_TXN_TRACE
FUNCTION_INVOCATION uses COMPUTATION_RESOURCES instead of EXECUTION_RESOURCES, but keeps the tag the same

File: rpc/types_transaction_receipt.go [hunk 1]
```diff
@@ -406,6 +406,19 @@ func (er *ExecutionResources) Validate() bool {
...
+	ComputationResources ComputationResources `json:"computation_resources"`
+	DataAvailability     DataAvailability     `json:"data_availability"`
+}
+
+type DataAvailability struct {
+	// the gas consumed by this transaction's data, 0 if it uses data gas for DA
+	L1Gas int `json:"l1_gas"`
+	// the data gas consumed by this transaction's data, 0 if it uses gas for DA
+	L1DataGas int `json:"l1_data_gas"`
+}
```
Comment (IronGauntlets) [inline rpc/types_transaction_receipt.go]: use `uint` instead of `int` as the gas consume cannot be negative

File: rpc/transaction_test.go [hunk 2]
```diff
@@ -208,14 +208,6 @@ func TestTransactionReceipt_MatchesCapturedTransaction(t *testing.T) {
...
 			},
 		},
-		ExecutionResources: ExecutionResources{
-			Steps:          217182,
-			MemoryHoles:    6644,
-			PedersenApps:   2142,
-			RangeCheckApps: 8867,
-			BitwiseApps:    900,
-			ECDSAApps:      1,
-		},
```
Comment (IronGauntlets) [inline rpc/transaction_test.go]: I don't understand why this is being removed. Should this not be replaced with `ComputationResources`?

Comment (IronGauntlets) [reply rpc/transaction_test.go]: Same for others

File: rpc/types_trace.go [hunk 3]
```diff
@@ -39,28 +39,31 @@ var _ TxnTrace = L1HandlerTxnTrace{}
...
 	//the trace of the __execute__ call or constructor call, depending on the transaction type (none for declare transactions)
-	ExecuteInvocation     ExecInvocation  `json:"execute_invocation"`
-	FeeTransferInvocation FnInvocation    `json:"fee_transfer_invocation"`
-	StateDiff             StateDiff       `json:"state_diff"`
-	Type                  TransactionType `json:"type"`
+	ExecuteInvocation     ExecInvocation     `json:"execute_invocation"`
+	FeeTransferInvocation FnInvocation       `json:"fee_transfer_invocation"`
+	StateDiff             StateDiff          `json:"state_diff"`
+	Type                  TransactionType    `json:"type"`
+	ExecutionResources    ExecutionResources `json:"execution_resources"`
```
Comment (IronGauntlets) [inline rpc/types_trace.go:46]: There is no usage or test that I can see for this addition.

**Review by IronGauntlets — CHANGES_REQUESTED** [review]

File: rpc/transaction_test.go (hunk 2 above)
Comment (rianhughes) [reply rpc/transaction_test.go]: Re-added

File: rpc/types_trace.go (hunk 3 above)
Comment (rianhughes) [reply rpc/types_trace.go:46]: Will try and get test data from Juno

Comment (rianhughes) [reply rpc/types_trace.go:46]: Later commit adds test

Comment (rianhughes) [reply rpc/types_trace.go:46]: TestInvokReceipt

Comment (rianhughes) [reply rpc/types_trace.go:46]: updated TestTransactionTrace to use a trace from RPCv07-rc2

File: rpc/trace_test.go [hunk 4]
```diff
@@ -77,7 +78,14 @@ func TestTransactionTrace(t *testing.T) {
...
 		} else {
-			require.Equal(t, test.ExpectedResp, resp)
+			invokeTrace := resp.(InvokeTxnTrace)
+			require.Equal(t, invokeTrace.ExecuteInvocation, test.ExpectedResp.ExecuteInvocation)
+			require.Equal(t, invokeTrace.Type, test.ExpectedResp.Type)
+			require.Equal(t, invokeTrace.StateDiff, test.ExpectedResp.StateDiff)
+			require.Equal(t, invokeTrace.FeeTransferInvocation, test.ExpectedResp.FeeTransferInvocation)
+			require.Equal(t, invokeTrace.ExecutionResources, test.ExpectedResp.ExecutionResources)
+			require.NotNil(t, invokeTrace.ExecutionResources)
+			fmt.Println(invokeTrace.ExecutionResources)
```
Comment (IronGauntlets) [inline rpc/trace_test.go]: If the fields and types of `invokeTrace` and `test`, I think `require.Equal` is sufficient.

File: rpc/transaction_test.go [hunk 5]
```diff
@@ -346,69 +350,59 @@ func TestTransactionReceipt_MatchesStatus(t *testing.T) {
...
-		}
+		txReceipt, err := testConfig.provider.TransactionReceipt(context.Background(), test.TxnHash)
+
+		require.Nil(t, err)
+		require.Equal(t, txReceipt.Hash(), test.ExpectedTxnReceipt.Hash())
+		require.Equal(t, txReceipt.GetExecutionStatus(), test.ExpectedTxnReceipt.GetExecutionStatus())
+		txReceiptInvoke, ok := txReceipt.(InvokeTransactionReceipt)
+		require.True(t, ok)
+		require.Equal(t, txReceiptInvoke.ActualFee, test.ExpectedTxnReceipt.ActualFee)
+		require.Equal(t, txReceiptInvoke.ExecutionResources, test.ExpectedTxnReceipt.ExecutionResources)
```
Comment (IronGauntlets) [inline rpc/transaction_test.go]: same as above

**Review by IronGauntlets — APPROVED** [review]
Approving as the changes proposed are trival

**Pull request merged**